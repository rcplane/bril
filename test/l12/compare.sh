#!/bin/zsh

echo "CS 6120 L12 dynamic compilers: trace executions of bril programs, stitch lvn dce and run checking correctness and dynamic instruction count"


echo " birthday 23 record trace"
echo " birthday 23 from trace dce lvn"
echo " birthday 22 from trace dce lvn miss"
echo " birthday 50 from trace dce lvn miss"
echo " and last three again default profiling" 
echo " "
echo " gcd 4 20 record trace"
echo " gcd 4 20 from trace dce lvn"
echo " gcd 4 80 from trace dce lvn ?hit"
echo " gcd 3 21 from trace dce lvn miss"
echo " and last three again default profiling"
echo " "
echo " perfect 496 record trace"
echo " perfect 496 from trace dce lvn"
echo " perfect 28  from trace dce lvn ?hit"
echo " perfect 12  from trace dce lvn miss"
echo " and last three again default profiling"

echo " prerequisite usual brili deno and haskell ghc and stack installs"
echo " https://github.com/rcplane/bril/tree/l12#reference-interpreter"
echo " https://github.com/rcplane/bril/tree/l12/bril-hs#usage "

set -x

bril2json < benchmarks/core/birthday.bril | brili -p -t benchmarks/core/birthday.trace 23
cat benchmarks/core/birthday.trace | stack --stack-yaml bril-hs/stack.yaml run -- --dce --lvn | brili -p -i benchmarks/core/birthday.args
bril2json < benchmarks/core/birthday.bril | brili -p -i benchmarks/core/birthday.args

bril2json < benchmarks/core/gcd.bril | brili -p -t benchmarks/core/gcd.trace 4 20
cat benchmarks/core/gcd.trace | stack --stack-yaml bril-hs/stack.yaml run -- --dce --lvn | brili -p -i benchmarks/core/gcd.args
bril2json < benchmarks/core/gcd.bril | brili -p -i benchmarks/core/gcd.args

bril2json < benchmarks/core/perfect.bril | brili -p -t benchmarks/core/perfect.trace 496  # todo head -n 1 args
cat benchmarks/core/perfect.trace | stack --stack-yaml bril-hs/stack.yaml run -- --dce --lvn | brili -p -i benchmarks/core/perfect.args
bril2json < benchmarks/core/perfect.bril | brili -p -i benchmarks/core/perfect.args

