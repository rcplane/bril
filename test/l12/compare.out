(base) b bril :$ ./test/l12/compare.sh  
CS 6120 L12 dynamic compilers: trace executions of bril programs, stitch lvn dce and run checking correctness and dynamic instruction count
 birthday 23 record trace
 birthday 23 from trace dce lvn
 birthday 22 from trace dce lvn miss
 birthday 50 from trace dce lvn miss
 and last three again default profiling dce lvn
 and last three again default profiling
 
 gcd 4 20 record trace
 gcd 4 20 from trace dce lvn
 gcd 4 80 from trace dce lvn ?hit
 gcd 3 21 from trace dce lvn miss
 and last three again default profiling dce lvn
 and last three again default profiling
 
 perfect 496 record trace
 perfect 496 from trace dce lvn
 perfect 28  from trace dce lvn ?hit
 perfect 12  from trace dce lvn miss
 and last three again default profiling dce lvn
 and last three again default profiling
 prerequisite usual brili deno and haskell ghc and stack installs
 https://github.com/rcplane/bril/tree/l12#reference-interpreter
 https://github.com/rcplane/bril/tree/l12/bril-hs#usage 
+./test/l12/compare.sh:33> bril2json
+./test/l12/compare.sh:33> brili -p -t benchmarks/core/birthday.trace 23
0.50729723432398566
total_dyn_inst: 484
+./test/l12/compare.sh:34> cat benchmarks/core/birthday.trace
+./test/l12/compare.sh:34> brili -p -i benchmarks/core/birthday.args
+./test/l12/compare.sh:34> stack --stack-yaml bril-hs/stack.yaml run -- --dce --lvn
0.50729723432398566
total_dyn_inst: 325
0.47569530766255030
total_dyn_inst: 311
0.97037357957798842
total_dyn_inst: 703
+./test/l12/compare.sh:35> bril2json
+./test/l12/compare.sh:35> stack --stack-yaml bril-hs/stack.yaml run -- --dce --lvn
+./test/l12/compare.sh:35> brili -p -i benchmarks/core/birthday.args
0.50729723432398566
total_dyn_inst: 254
0.47569530766255030
total_dyn_inst: 243
0.97037357957798842
total_dyn_inst: 551
+./test/l12/compare.sh:36> bril2json
+./test/l12/compare.sh:36> brili -p -i benchmarks/core/birthday.args
0.50729723432398566
total_dyn_inst: 484
0.47569530766255030
total_dyn_inst: 463
0.97037357957798842
total_dyn_inst: 1051
+./test/l12/compare.sh:38> bril2json
+./test/l12/compare.sh:38> brili -p -t benchmarks/core/gcd.trace 4 20
4
total_dyn_inst: 46
+./test/l12/compare.sh:39> cat benchmarks/core/gcd.trace
+./test/l12/compare.sh:39> stack --stack-yaml bril-hs/stack.yaml run -- --dce --lvn
+./test/l12/compare.sh:39> brili -p -i benchmarks/core/gcd.args
4
total_dyn_inst: 63
4
total_dyn_inst: 243
3
total_dyn_inst: 87
+./test/l12/compare.sh:40> bril2json
+./test/l12/compare.sh:40> stack --stack-yaml bril-hs/stack.yaml run -- --dce --lvn
+./test/l12/compare.sh:40> brili -p -i benchmarks/core/gcd.args
4
total_dyn_inst: 46
4
total_dyn_inst: 181
3
total_dyn_inst: 64
+./test/l12/compare.sh:41> bril2json
+./test/l12/compare.sh:41> brili -p -i benchmarks/core/gcd.args
4
total_dyn_inst: 46
4
total_dyn_inst: 181
3
total_dyn_inst: 64
+./test/l12/compare.sh:43> bril2json
+./test/l12/compare.sh:43> brili -p -t benchmarks/core/perfect.trace 496
0
total_dyn_inst: 232
+./test/l12/compare.sh:44> cat benchmarks/core/perfect.trace
+./test/l12/compare.sh:44> stack --stack-yaml bril-hs/stack.yaml run -- --dce --lvn
+./test/l12/compare.sh:44> brili -p -i benchmarks/core/perfect.args
0
total_dyn_inst: 371
0
total_dyn_inst: 98
1
total_dyn_inst: 67
+./test/l12/compare.sh:45> bril2json
+./test/l12/compare.sh:45> stack --stack-yaml bril-hs/stack.yaml run -- --dce --lvn
+./test/l12/compare.sh:45> brili -p -i benchmarks/core/perfect.args
0
total_dyn_inst: 231
0
total_dyn_inst: 57
1
total_dyn_inst: 36
+./test/l12/compare.sh:46> bril2json
+./test/l12/compare.sh:46> brili -p -i benchmarks/core/perfect.args
0
total_dyn_inst: 232
0
total_dyn_inst: 58
1
total_dyn_inst: 37
